internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -3262,6 +3319,24 @@ func flattenKubernetesClusterWorkloadAutoscalerProfile(profile *managedclusters.
 	}
 }
 
+func flattenKubernetesClusterNodeProvisioningProfile(profile *managedclusters.ManagedClusterNodeProvisioningProfile) []interface{} {
+	if profile == nil || (profile.Mode == nil && profile.DefaultNodePools == nil) {
```

Comment:
What is the difference to always setting this whole block to state even if all the attributes are just nil?